### PR TITLE
BAI-1299 Python semantic versioning bugfix

### PR DESCRIPTION
**Pull request by bw27492**
- Semantic versons containing 'v' will no longer break client

### CODE REVIEW

File: lib/python/src/bailo/helper/release.py [hunk 1]
```diff
@@ -275,16 +273,33 @@ def delete(self) -> Any:
...
         return True
 
+    @property
+    def version(self):
+        return self._version_obj
+
+    @version.setter
+    def version(self, value):
+        if isinstance(value, str):
+            version_obj = value.replace("v", "")
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/release.py:283]: Done

File: lib/python/src/bailo/helper/release.py [hunk 2]
```diff
@@ -275,16 +273,33 @@ def delete(self) -> Any:
...
+            version_obj = Version.coerce(version_obj)
+        elif isinstance(value, Version):
+            version_obj = value
+        else:
+            raise TypeError("Provided version not of a supported type.")
+
+        self._version_obj = version_obj
+        self._version_raw = value
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/release.py:291]: Done

File: lib/python/src/bailo/helper/release.py [hunk 3]
```diff
@@ -275,16 +273,33 @@ def delete(self) -> Any:
         self.client.delete_release(self.model_id, str(self.version))
         return True
 
+    @property
+    def version(self):
+        return self._version_obj
+
+    @version.setter
+    def version(self, value):
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/release.py:281]: done

[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/release.py:281]: I don't think we want to fully stop users from adding the 'v' using the `post_release` method, as the front and backend will allow this. It only presented an issue when creating the `Version()` object which is used to compare `Release()` objects etc, which is resolved with this PR whilst allowing the user to include the 'v' upon creation if they want to

File: lib/python/src/bailo/helper/release.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/release.py:291]: The `_version_raw` attribute is the same value used to get/create releases on the server - so if the user inputs an incorrect value the server will return an error anyway. Also if we get the semver from the object, the `v` will have been removed so if the user calls `.update()` the server won't find the right release (e.g. the client will try to get 2.0.0 as opposed to v2.0.0)

[the remark this answers is not in the document]
Comment (ARADDCC002) [reply lib/python/src/bailo/helper/release.py:292]: > The `_version_raw` attribute is the same value used to get/create releases on the server - so if the user inputs an incorrect value the server will return an error anyway. Also if we get the semver from the object, the `v` will have been removed so if the user calls `.update()` the server won't find the right release (e.g. the client will try to get 2.0.0 as opposed to v2.0.0)



> The `_version_raw` attribute is the same value used to get/create releases on the server - so if the user inputs an incorrect value the server will return an error anyway. Also if we get the semver from the object, the `v` will have been removed so if the user calls `.update()` the server won't find the right release (e.g. the client will try to get 2.0.0 as opposed to v2.0.0)

test

File: lib/python/src/bailo/helper/release.py [hunk 4]
```diff
@@ -275,16 +273,34 @@ def delete(self) -> Any:
...
```
Comment (GB27247) [inline lib/python/src/bailo/helper/release.py:281]: I think it's best to make this private. Just to avoid any problematic updating.

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: lib/python/src/bailo/helper/release.py (hunk 4 above)
Comment (bw27492) [reply lib/python/src/bailo/helper/release.py:281]: Done

**Review by GB27247 — APPROVED** [review]

**Pull request merged**